pylint/checkers/variables.py

```diff
@@ -1326,9 +1327,35 @@ def visit_for(self, node: nodes.For) -> None:
             if any(isinstance(target, nodes.Starred) for target in targets):
                 return
 
-        if len(targets) != len(values):
-            details = _get_unpacking_extra_info(node, inferred)
-            self._report_unbalanced_unpacking(node, inferred, targets, values, details)
+        if isinstance(inferred, nodes.Dict) and isinstance(node.iter, nodes.Name):
+            # If this a case of 'dict-items-missing-iter', we don't want to
+            # report it as an 'unbalanced-dict-unpacking' as well
+            # TODO (performance), merging both checks would streamline this
+            if len(targets) == 2:
+                return
+
+        if isinstance(
+            inferred,
+            (
+                astroid.objects.DictKeys,
+                astroid.objects.DictValues,
+                astroid.objects.DictItems,
+            ),
+        ):
+            is_stared_targets = any(
```

Comment:
```suggestion
        if isinstance(inferred, nodes.Dict):
            if isinstance(node.iter, nodes.Name):
                # If this a case of 'dict-items-missing-iter', we don't want to
                # report it as an 'unbalanced-dict-unpacking' as well
                # TODO (performance), merging both checks would streamline this
                if len(targets) == 2:
                    return

        else:
            is_stared_targets = any(
```
Perhaps a bit better like this since `inferred` is one of Dict, DictKeys, DictValues, DictItems.

Reply:
Yeah, that reads better, and passes tests. Committed.